src/tasks/withdraw/traded_tokens.ts

```diff
@@ -8,6 +8,16 @@ const enum ProviderError {
   Timeout,
 }
 
+export interface TradedTokens {
+  tokens: string[];
+  toBlock: number;
+}
+
+export const partialTradedTokensKey = "partialTradedTokens" as const;
+export interface TradedTokensError extends Error {
+  [partialTradedTokensKey]: TradedTokens;
```

Comment:
The partial tokens were already exposed before, we just didn't use them?

Reply:
Tokens are retrieved by checking all events from the first to the last block, so this information was in a way always there in a way, it was just a matter of exposing it in a nice way.